### PR TITLE
antsimulator: init at 1.2

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/antsimulator/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "v${version}";
+    sha256 = "0wz80971rf86kb7mcnxwrq75vriwhmyir5s5n3wzml12rzfnj5f1";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ sfml ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./AntSimulator $out/bin
```
Comment (SuperSandro2000) [inline pkgs/games/antsimulator/default.nix:19]: Maybe the binary should be lowercased.

Comment (IvarWithoutBones) [reply pkgs/games/antsimulator/default.nix]: Sure, done :+1:

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118636` at a2460b8d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>antsimulator</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/games/antsimulator/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by voidIess** [comment]
The sim started built and started without problems. This was stdout. Dont know if it is supposed to be like this.

```
Failed to load image "res/ant_2.png". Reason: Unable to open file
Failed to load image "res/marker.png". Reason: Unable to open file
Couldn't find 'conf.txt', loading default
Failed to load image "res/circle.png". Reason: Unable to open file
```

And the sim didnt automaticly update when changing window size. But thats propably intendet.

File: pkgs/games/antsimulator/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "v${version}";
+    sha256 = "0wz80971rf86kb7mcnxwrq75vriwhmyir5s5n3wzml12rzfnj5f1";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ sfml ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./AntSimulator $out/bin/antsimulator
```
Comment (sikmir) [inline pkgs/games/antsimulator/default.nix:19]:
```suggestion
    install -Dm755 ./AntSimulator $out/bin/antsimulator
```

Comment (IvarWithoutBones) [reply pkgs/games/antsimulator/default.nix]: Done, thanks :+1:

**Comment by IvarWithoutBones** [comment]
> The sim started built and started without problems. This was stdout. Dont know if it is supposed to be like this.
> 
> ```
> Failed to load image "res/ant_2.png". Reason: Unable to open file
> Failed to load image "res/marker.png". Reason: Unable to open file
> Couldn't find 'conf.txt', loading default
> Failed to load image "res/circle.png". Reason: Unable to open file
> ```
> 
> And the sim didnt automaticly update when changing window size. But thats propably intendet.

I saw those errors as well, I've tried putting both the `res` and the executable in `$out/lib`, and than using `makeWrapper` to symlink this to `$out/bin`. This unfortunately didn't fix the problem. 

I'm unsure what it effects, as the program seems to be working as intended.

**Pull request merged**